plugins/chat/app/services/chat/lookup_user_threads.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module Chat
+  # Gets a list of threads for a user.
+  #
+  # Only threads that the user is a member of with a notification level
+  # of normal or tracking will be returned.
+  #
+  # @example
+  #  Chat::LookupUserThreads.call(guardian: guardian, limit: 5, offset: 2)
+  #
+  class LookupUserThreads
+    include Service::Base
+
+    THREADS_LIMIT = 10
+
+    # @!method call(guardian:, limit: nil, offset: nil)
+    #   @param [Guardian] guardian
+    #   @param [Integer] limit
+    #   @param [Integer] offset
+    #   @return [Service::Base::Context]
+
+    contract
+    step :set_limit
+    step :set_offset
+    model :threads
+    step :fetch_tracking
+    step :fetch_memberships
+    step :fetch_participants
+    step :build_load_more_url
+
+    # @!visibility private
+    class Contract
+      attribute :limit, :integer
+      attribute :offset, :integer
+    end
+
+    private
+
+    def set_limit(contract:, **)
+      context.limit = (contract.limit || THREADS_LIMIT).to_i.clamp(1, THREADS_LIMIT)
+    end
+
+    def set_offset(contract:, **)
+      context.offset = [contract.offset || 0, 0].max
+    end
+
+    def fetch_threads(guardian:, **)
+      ::Chat::Thread
+        .includes(
+          :channel,
+          :user_chat_thread_memberships,
+          original_message_user: :user_status,
+          last_message: [
+            :uploads,
+            :chat_webhook_event,
+            :chat_channel,
+            chat_mentions: {
+              user: :user_status,
+            },
+            user: :user_status,
+          ],
+          original_message: [
+            :uploads,
+            :chat_webhook_event,
+            :chat_channel,
+            chat_mentions: {
+              user: :user_status,
+            },
+            user: :user_status,
+          ],
+        )
+        .joins(
+          "INNER JOIN user_chat_thread_memberships ON chat_threads.id = user_chat_thread_memberships.thread_id",
+        )
+        .joins(
+          "LEFT JOIN chat_messages AS last_message ON chat_threads.last_message_id = last_message.id",
+        )
+        .joins(
+          "INNER JOIN chat_messages AS original_message ON chat_threads.original_message_id = original_message.id",
+        )
+        .where(
+          channel_id:
+            ::Chat::Channel
+              .joins(:user_chat_channel_memberships)
+              .where(user_chat_channel_memberships: { user_id: guardian.user.id })
+              .where(
+                {
+                  chatable_type: "Category",
+                  threading_enabled: true,
+                  status: ::Chat::Channel.statuses[:open],
+                },
+              )
+              .select(:id),
```

Comment:
I think that `pluck(:id)` may be slightly better here? For membership, do you want to exclude channels the user has `muted: true` or where `following: false` here too?

Reply:
Can you have a thread following=false? how do you do that in the UI?

Reply:
No but you can have a `UserChatChannelMembership` that you are not following or have muted, so probably don't even need to pluck channel ID for those ones. So there extra where would be like:

```ruby
.where("user_chat_channel_memberships.following AND NOT user_chat_channel_memberships.muted")

Reply:
yes seems right